### PR TITLE
Initial implementation of ScrollSync Service.

### PR DESCRIPTION
**Pull request by mkhatib**

### CODE REVIEW

File: extensions/amp-fx-scroll-sync/0.1/scroll-sync-service.js [hunk 1]
```diff
@@ -14,29 +14,99 @@
...
+    this.win_ = win;
+
     /** @private @const {!Array<!ScrollSyncEffect>} */
     this.effects_ = [];
+
+    this.viewport_ = viewportFor(win);
+    this.vsync_ = vsyncFor(win);
+
+    this.viewport_.onScroll(() => {
+      this.vsync_.run({
```
Comment (dvoytenko) [inline extensions/amp-fx-scroll-sync/0.1/scroll-sync-service.js]: Scroll top is generally allowed to be read right from `onScroll`. But, of course, vsync is great as well. Just FYI.

File: extensions/amp-fx-scroll-sync/0.1/scroll-sync-service.js [hunk 2]
```diff
@@ -14,29 +14,99 @@
...
+
+
+function measureScrollTop(state) {
+  state.scrollTop = state.viewport.getScrollTop();
+}
+
 
+function onScroll(state) {
+  const scrollTop = state.scrollTop;
+  for (const effect of state.effects) {
```
Comment (jridgewell) [inline extensions/amp-fx-scroll-sync/0.1/scroll-sync-service.js]: Eek! This imports a gigantic polyfill.

Comment (dvoytenko) [inline extensions/amp-fx-scroll-sync/0.1/scroll-sync-service.js]: @jridgewell These are just samples - not a prod code.

Comment (erwinmombay) [inline extensions/amp-fx-scroll-sync/0.1/scroll-sync-service.js]: looks like we currently don't. adding lint

Comment (erwinmombay) [inline extensions/amp-fx-scroll-sync/0.1/scroll-sync-service.js]: stand corrected, lint is in place. just forgot

Comment (muxin) [inline extensions/amp-fx-scroll-sync/0.1/scroll-sync-service.js]: This is fixed in https://github.com/ampproject/amphtml/pull/4530

**Pull request merged**